### ISSUE TITLE
"modes of use" doc page

### ISSUE THREAD

**Issue by martindurant** [intake/intake#397]
It would be nice to create a page on the documentation listing some of the ways in which Intake is deployed, including:

- simple YAML files being passed around
- parsing simple responses from a server to locate resources (e.g., intake-thredds)
- scanning directory structures and extracting information into catalogs (e.g., intake-esm @jukent @andersy005 )
- performing queries on a remote server to produce totally dynamic catalogs (e.g., intake-bluesky @danielballan )
- totally arbitrary functionality in a server

**Comment by danielballan** [comment]
Yes, I think making these other uses beyond "simple YAML files" more prominent in the documentation would be helpful for selling intake more broadly. Sign me up for contributing some content. Can you aim us a skeleton of a document to start with?

**Comment by andersy005** [comment]
I'd be happy to contribute some content as well.

**Comment by talebzeghmi** [comment]
I'd be happy to contribute on how we use Intake at Zillow too!
